crypto_secretstream/src/header.rs

```diff
@@ -0,0 +1,81 @@
+use std::convert::TryFrom;
+
+use chacha20::cipher::{
+    consts::U24,
+    generic_array::{sequence::Split, GenericArray},
+};
+use rand_core::{CryptoRng, RngCore};
+use snafu::ensure;
+
+use super::nonce::{HChaCha20Nonce, Nonce};
+
+#[derive(Debug, snafu::Snafu)]
+pub enum Error {
+    InvalidLength,
+}
+
+/// Header of the secret stream, can be sent as cleartext.
+#[derive(Clone, Copy)]
+pub struct Header(GenericArray<u8, U24>);
+
+impl Header {
+    /// Number of bytes used by the serialisation.
+    pub const BYTES: usize = 24;
+
+    /// Generate a new random [`Header`].
+    pub(super) fn generate<T>(csprng: &mut T) -> Self
+    where
+        T: RngCore + CryptoRng,
```

Comment:
We've generally used `impl Trait` for this elsewhere to have a slightly more readable signature:

```suggestion
    pub(super) fn generate(csprng: impl CryptoRng + RngCore) -> Self
```